addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -143,50 +159,112 @@ private void changeState(final InternalLedState newState) throws IOException {
         }
     }
 
-    private void fadeToState(final InternalLedState newState) throws IOException {
-        if (!newState.equals(realTargetState)) {
-            keepFading = false;
-            realTargetState = newState;
+    final static class LEDFaderRunner implements Runnable {
+        protected Logger logger = LoggerFactory.getLogger(LEDFaderRunner.class);
+        InternalLedState fadeState;
+        int fadeStepper = 1;
+
+        final InternalLedState toState;
+        final InternalLedState fromState;
+        final Future<?>[] faderArray;
+        final int steps;
+        final Function<InternalLedState, Boolean> ledSender;
+
+        public LEDFaderRunner(Future<?>[] faderFuture, InternalLedState fromState, InternalLedState toState, int steps, Function<InternalLedState,  Boolean> ledSender) {
+            this.faderArray = faderFuture;
+            fadeState=fromState;
```

Comment:
Martin,

I looked around on the internet and in the source code but I have trouble finding the Eclipse ESH profile.  Can you kindly point me in the right direction? At this moment I use the default Idea formatting without formatting of wildcart imports. If it's the default from Eclipse, I can look into that and see if I can get that matching in Idea.